.circleci/config.yml

```diff
@@ -37,19 +37,45 @@ jobs:
+                               | grep -v '@Nested' \
                                | xargs grep -l '@Test' \
+                               | xargs grep -L '@Nested' \
                                | sed 's/.*src.test.java.//g' | sed 's@/@.@g' \
-                               | sed 's/.\{5\}$//' \
-                               | circleci tests split --split-by=timings)
-            echo $CLASSNAMES
-            TEST_ARG=$(echo $CLASSNAMES | sed 's/ /,/g')
-            echo $TEST_ARG
-            ./mvnw -s .settings.xml  -DfailIfNoTests=false -DtestsToRun=$TEST_ARG -e clean org.jacoco:jacoco-maven-plugin:prepare-agent test -U -P sonar -nsu --batch-mode -Dmaven.test.redirectTestOutputToFile=true -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+                               | sed 's/.\{5\}$//' > /tmp/all_tests.txt
+
+            # find classnames that have @Nested inside them
+            ROOT_NESTED_CLASSNAMES=$(circleci tests glob "**/src/test/**/**.java" \
+                               | grep -v 'spring-cloud-kubernetes-integration-tests' \
+                               | xargs grep -l '@Nested' \
+                               | xargs grep -l '@Test')
+            declare -a 'ARRAY_ROOT_NESTED_CLASSNAMES=($ROOT_NESTED_CLASSNAMES)'
+            for p in "${ARRAY_ROOT_NESTED_CLASSNAMES[@]}"; do
+            	# find inner @Nested class names
+                flattened=$(cat $p | awk '/@Nested/, /@Test/' | tr '\n' ' ' | awk '{ for (i = 1; i <= NF; ++i) { if( $i == "class") { print $(i+1) } } }')
+                declare -a 'flat_array=($flattened)'
+            	for flat in "${flat_array[@]}"; do
+            		x=$(echo $p | sed 's/.*src.test.java.//g' | sed 's@/@.@g' | sed 's/.\{5\}$//')
+            		y="$x\$"$flat""
+            		echo -e $y >> /tmp/all_tests.txt
+            	done
+            done
+            cat /tmp/all_tests.txt | circleci tests split --split-by=timings > /tmp/this_node_tests.txt
+            tr '\n' ',' < /tmp/this_node_tests.txt > /tmp/this_node_tests_comma_separated.txt
+            sed -i 's/\\$/\\\\$/' /tmp/this_node_tests_comma_separated.txt
+
+            echo '------- tests for this node ----------'
+            cat /tmp/this_node_tests_comma_separated.txt
+            echo -e '\n--------------------------------------'
+            ./mvnw -s .settings.xml  -DfailIfNoTests=false -DtestsToRun=$(cat /tmp/this_node_tests_comma_separated.txt) -e clean org.jacoco:jacoco-maven-plugin:prepare-agent test -U -P sonar -nsu --batch-mode -Dmaven.test.redirectTestOutputToFile=true -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
             mkdir -p $HOME/artifacts/junit/
             find . -type f -regex ".*/spring-cloud-*.*/target/*.*" -exec cp {} $HOME/artifacts/ \;
             find . -type f -regex ".*/target/.*-reports/.*" -exec cp {} $HOME/artifacts/junit/ \;
-            bash <(curl -s https://codecov.io/bash)
```

Comment:
this one is deprecated and their page says that:

```
"On February 1, 2022, this uploader will be fully sunset and no longer function"
```

the recommendation is to use [this one](https://github.com/codecov/uploader), but it requires `npm` and says "Heads up! This is a work in progress.". We will need to investigate a bit more on what to use and how.